src/me/mrCookieSlime/Slimefun/api/HashedArmorpiece.java

```diff
@@ -0,0 +1,38 @@
+package me.mrCookieSlime.Slimefun.api;
+
+import java.util.Optional;
+
+import org.bukkit.inventory.ItemStack;
+
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunArmorPiece;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+
+public final class HashedArmorpiece {
+	
+	private int hash;
+	private Optional<SlimefunArmorPiece> item;
```

Comment:
What kind of reference will this be? Maybe it should be enforced as a weak reference?

Reply:
Why?

Reply:
If this is held elsewhere then the reference should be controlled. Not sure if it is or not though

Reply:
Well, only the class itself access it, it does have a getter but the entire PlayerProfile will be unloaded from memory when the player leaves anyway.